OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -260,17 +260,20 @@ public HHNetworkRouteRes runRouting(LatLon start, LatLon end, HHRoutingConfig co
 		return route;
 	}
 
-	private void findFirstLastSegments(HHRoutingContext<T> hctx, LatLon start, LatLon end, 
-			 TLongObjectHashMap<T> stPoints, TLongObjectHashMap<T> endPoints) throws IOException, InterruptedException {
+	private void findFirstLastSegments(HHRoutingContext<T> hctx, LatLon start, LatLon end,
+	                                   TLongObjectHashMap<T> stPoints, TLongObjectHashMap<T> endPoints,
+	                                   RouteCalculationProgress progress) throws IOException, InterruptedException {
 		long time = System.nanoTime();
 		System.out.println("Finding first / last segments...");
 		RoutePlannerFrontEnd planner = new RoutePlannerFrontEnd();
 		int startReiterate = -1, endReiterate = -1;
 		boolean found = false;
+		progress.hhIterationProgress(0.00);
```

Comment:
Use HHIteration

Reply:
progress.hhIterationProgress is not about fixed enumeration. it is `double` (%) for current step